src/app/domain/sharing/services/SharingService.ts

```diff
@@ -0,0 +1,75 @@
+import ReceiveSharingIntent from '@mythologi/react-native-receive-sharing-intent'
+
+import Minilog from 'cozy-minilog'
+
+import {
+  isAndroidFile,
+  isIOSFile,
+  ReceivedFile
+} from '/app/domain/sharing/models/ReceivedFile'
+
+export const sharingLogger = Minilog('🗃️ SharingService')
+const filesToUpload = new Map<string, ReceivedFile>()
+
+// Add files to upload in a Map to avoid duplicates created by a bug in the library
+// Map key key depends on the platform, typeguard allows us to benefit from type inference
+const processReceivedFiles = (files: ReceivedFile[]): void => {
+  sharingLogger.info('Processing received files', files)
+
+  // We don't want to keep files from a previous sharing session
+  if (filesToUpload.size > 0) {
+    sharingLogger.info('Clearing files to upload')
+    filesToUpload.clear()
+  }
+
+  // Populate the `filesToUpload` Map with the files to upload
+  for (const file of files) {
+    let key: string | undefined | null
+
+    if (isAndroidFile(file)) {
+      key = file.filePath
+    } else if (isIOSFile(file)) {
```

Comment:
There seems to be no difference between iOS and Android? Bot do `key = file.filePath`